hydra/core/override_parser/overrides_visitor.py

```diff
@@ -240,6 +249,44 @@ def visitFunction(self, ctx: OverrideParser.FunctionContext) -> Any:
                 f"{type(e).__name__} while evaluating '{ctx.getText()}': {e}"
             ) from e
 
+    def visitQuotedValue(self, ctx: OverrideParser.QuotedValueContext) -> QuotedString:
+        children = list(ctx.getChildren())
+        assert len(children) >= 2
+
+        # Single or double quote?
+        first_quote = children[0].getText()
+        if first_quote == "'":
+            quote = Quote.single
+        else:
+            assert first_quote == '"'
+            quote = Quote.double
+
+        tokens = []
+        is_interpolation = False
+        for child in children[1:-1]:  # iterate on child nodes between quotes
+            if isinstance(child, TerminalNode):
+                s = child.symbol
+                if s.type == OverrideLexer.ESC_QUOTE:
+                    # Always un-escape quotes.
+                    tokens.append(s.text[1])
+                    continue
+                if s.type == OverrideLexer.ESC_INTER:
+                    # OmegaConf processes escaped interpolations as interpolations.
+                    is_interpolation = True
+            else:
+                assert isinstance(child, OverrideParser.InterpolationContext)
+                is_interpolation = True
+            tokens.append(child.getText())
+
+        ret = "".join(tokens)
+
+        # If it is an interpolation, then OmegaConf will take care of un-escaping
+        # the `\\`. But if it is not, then we need to do it here.
+        if not is_interpolation:
+            ret = ret.replace("\\\\", "\\")
```

Comment:
I wanted to discuss this point. It looks (and is) a bit clumsy but it's the best I have right now.

Consider the two following override values (here we really have 4 backslashes ending each quoted string, I'm assuming these are the actual characters after any Python un-escaping):
1. `["abc\\\\", "def\\\\"]`
2. `["${abc}\\\\", "${def}\\\\"]`

In both situations we would like the `\\\\` to be turned into `\\` (i.e., `\\\\` == two consecutive escaped backslashes).

In case (1), if the content of each quoted string is kept unchanged (ex: `abc\\\\`) then we'll end up with extra unwanted backslashes at the end of the string (since the intent here is to obtain `abc\\`).

In case (2), if Hydra un-escapes the backslahes, then it will feed the string `${abc}\\` to OmegaConf, which will resolve it to `<value_of_abc>\` (it will un-escape `\\` into `\`), so we would end up with only one `\` at the end instead of two.

As a result, we need to know whether or not OmegaConf will process the string with its grammar, in order to decide whether or not to un-escape the `\\`. I guess we could decide to ignore this and let the user figure it out, but this definitely doesn't seem intuitive.

@omry how do you feel about this?

Reply:
Note: this is a side effect of the fact that in OmegaConf, setting a value node to `abc\\` will use this string unchanged, while `${abc}\\` will be resolved to `<value_of_abc>\`.
One potential approach to resolve this issue would be the one from https://github.com/omry/omegaconf/pull/621 where `\\` is only un-escaped when required.